controllers/statussync/policy_status_sync.go

```diff
@@ -721,6 +735,24 @@ func StartComplianceEventsSyncer(
 	}
 }
 
+// getHubComplianceAPIToken retrieves the token associated with the service account with compliance history API
+// recording permssions in the cluster namespace.
+func getHubComplianceAPIToken(ctx context.Context, hubCfg *rest.Config, clusterNamespace string) (string, error) {
+	client, err := kubernetes.NewForConfig(hubCfg)
+	if err != nil {
+		return "", err
+	}
+
+	saTokenSecret, err := client.CoreV1().Secrets(clusterNamespace).Get(
+		ctx, hubComplianceAPISAName, metav1.GetOptions{},
+	)
+	if err != nil {
+		return "", err
+	}
+
+	return string(saTokenSecret.Data["token"]), nil
```

Comment:
Is there any case `saTokenSecret` is empty?

Reply:
Yeah, it could be empty in between the secret getting created and the secret getting populated with the token by Kubernetes.

I addressed this in the latest push. Good catch!